template.rb

```diff
@@ -40,31 +32,44 @@ def apply_template!
   git :init unless preexisting_git_repo?
   empty_directory ".git/safe"
 
-  run_with_clean_bundler_env "bundle update"
-  run_with_clean_bundler_env "bin/rails webpacker:install"
-  install_dart_sass unless sprockets?
-  create_database_and_initial_migration
-  run_with_clean_bundler_env "bin/setup"
-
-  binstubs = %w[brakeman bundler bundler-audit rubocop sidekiq]
-  run_with_clean_bundler_env "bundle binstubs #{binstubs.join(' ')} --force"
-
-  template "rubocop.yml.tt", ".rubocop.yml"
-  run_rubocop_autocorrections
-
-  template "eslintrc.js", ".eslintrc.js"
-  template "prettierrc.js", ".prettierrc.js"
-  template "stylelintrc.js", ".stylelintrc.js"
-  add_yarn_lint_and_run_fix
-  add_yarn_start_script
-
-  unless any_local_git_commits?
-    git checkout: "-b main"
-    git add: "-A ."
-    git commit: "-n -m 'Set up project'"
-    if git_repo_specified?
-      git remote: "add origin #{git_repo_url.shellescape}"
-      git push: "-u origin --all"
+  after_bundle do
+    apply "app/assets/template.rb"
+    apply "app/template.rb"
+
+    run_with_clean_bundler_env "bundle update"
+    create_database_and_initial_migration
+    run_with_clean_bundler_env "bin/setup"
+
+    binstubs = %w[brakeman bundler bundler-audit rubocop sidekiq]
+    run_with_clean_bundler_env "bundle binstubs #{binstubs.join(' ')} --force"
+
+    if File.exist?("bin/dev")
+      append_to_file "Procfile.dev", "worker: bin/sidekiq"
+    else
+      remove_file "Procfile.dev"
+    end
+
+    template "rubocop.yml.tt", ".rubocop.yml"
+    run_rubocop_autocorrections
+
+    template "eslintrc.js", ".eslintrc.js"
+    template "prettierrc.js", ".prettierrc.js"
+    template "stylelintrc.js", ".stylelintrc.js"
+    add_yarn_lint_and_run_fix
+    add_yarn_start_script
+
+    append_to_file ".gitignore", "node_modules" unless File.read(".gitignore").match?(%{^/?node_modules})
+
+    run_with_clean_bundler_env "bundle lock --add-platform x86_64-linux"
```

Comment:
Newer versions of Bundler will generate a Gemfile.lock that only works on the platform it was originally generated on (e.g. Mac). So if we want to be able to build on CI or deploy to Heroku, which are Linux, we have to explicitly opt in.